analyzers/tests/SonarAnalyzer.UnitTest/Rules/UseTrueForAllTest.cs

```diff
@@ -0,0 +1,50 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using CS = SonarAnalyzer.Rules.CSharp;
+using VB = SonarAnalyzer.Rules.VisualBasic;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class UseTrueForAllTest
+{
+    private readonly VerifierBuilder builderCS = new VerifierBuilder<CS.UseTrueForAll>();
+    private readonly VerifierBuilder builderVB = new VerifierBuilder<VB.UseTrueForAll>();
+
+    [TestMethod]
+    public void UseTrueForAll_CS() =>
+        builderCS.AddPaths("UseTrueForAll.cs").WithOutputKind(OutputKind.DynamicallyLinkedLibrary).Verify();
+
+#if NET
+
+    [TestMethod]
+    public void UseTrueForAll_CS_Immutable() =>
+        builderCS.AddPaths("UseTrueForAll.Immutable.cs")
+            .AddReferences(MetadataReferenceFacade.SystemCollections)
+            .WithOutputKind(OutputKind.DynamicallyLinkedLibrary)
```

Comment:
Educational: what is the purpose of this?

Reply:
Actually, I'm not sure, all of those UTs are running fine without this line. I think you need the "single line of code" only using `WithTopLevelStatements` where the compiler actually needs an entry point, but I think the functionality you mentioned is the default one

Reply:
You are absolutely right, the default behavior of the builder is `DynamicLinkedLibrary`. I did not see it because before I tried with top level statements, which as you said need at least a line of code. Thanks!